src/components/OneUser.tsx

```diff
@@ -0,0 +1,23 @@
+import { User } from '../types/User';
+
+type Props = {
+  user: User;
+  setSelectedUserId: (id: number | null) => void;
+  setIsDropDownOpen: (status : boolean) => void;
+};
+
+export const OneUser: React.FC<Props> = ({ user, setSelectedUserId, setIsDropDownOpen }) => {
+  const handleUserSelect = () => {
+    setSelectedUserId(user.id);
+    setIsDropDownOpen(false);
+  }
+  return (
+    <a
+      href={`#user-${user.id}`}
+      className="dropdown-item"
+      onClick={handleUserSelect}
+    >
+      {user.name}
```

Comment:
The `href` attribute in the anchor tag is using a template string to dynamically set the URL fragment. Ensure that the `user.id` is always defined to avoid potential issues.